app/root.tsx

```diff
@@ -7,13 +9,44 @@ import {
 	Scripts,
 	ScrollRestoration,
 	isRouteErrorResponse,
+	json,
 	useRouteError,
+	useRouteLoaderData,
 } from "@remix-run/react";
-
-import "@fontsource-variable/plus-jakarta-sans";
 import { ReactNode } from "react";
+import { fetchReports } from "~/server/impactReportHelpers";
 import "./tailwind.css";
 
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+	const ownerAddress = process.env.HC_OWNER_ADDRESS;
+	try {
+		if (!ownerAddress) {
+			throw new Error("Owner address environment variable is not set");
+		}
+		const reports = await fetchReports(ownerAddress);
+		// remix shortcut to return json instead of a string
+		return json(reports);
+		// return new Response((reports), {
+		// 	status: 200,
+		// 	statusText: "OK",
+		// });
+	} catch (error) {
+		console.error(`Failed to load impact reports: ${error}`);
+		return new Response(
+			JSON.stringify({ error: "Failed to load impact reports" }),
+			{
+				status: 500,
+				statusText: "Internal Server Error",
+			},
+		);
+	}
+};
+
+// custom hook to pass the loader's type to other routes
+export function useRootLoaderData() {
```

Comment:
no need to export a custom hook. in any other route just use `useRouteLoaderData<typeof loader>("root")` and it will work. this custom hook would cause confusion as it hijacks the original hook.